diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2eba16da5..ba0d28fa2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -283,11 +283,6 @@ if(LIB_INSTALL_DIR)
     set(CMAKE_INSTALL_LIBDIR "${LIB_INSTALL_DIR}")
 endif()

-if (NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/framework/CMakeLists.txt")
-    message(FATAL_ERROR "${CMAKE_CURRENT_SOURCE_DIR}/framework/CMakeLists.txt not found. Run `git submodule update --init` from the source tree to fetch the submodule contents.")
-endif()
-add_subdirectory(framework)
-
 add_subdirectory(include)

 add_subdirectory(3rdparty)
