From ee1780f9595c17aec2e7b1f2a67e03f8b069d4e1 Mon Sep 17 00:00:00 2001
From: Valerio Setti <valerio.setti@nordicsemi.no>
Date: Wed, 15 May 2024 07:39:47 +0200
Subject: [PATCH] config_psa: do not update legacy symbols in client-only PSA
 build

Signed-off-by: Valerio Setti <valerio.setti@nordicsemi.no>
---
 include/mbedtls/config_psa.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/mbedtls/config_psa.h b/include/mbedtls/config_psa.h
index 17da61b3e8..8bcca407fd 100644
--- a/include/mbedtls/config_psa.h
+++ b/include/mbedtls/config_psa.h
@@ -32,7 +32,11 @@
  * before we deduce what built-ins are required. */
 #include "psa/crypto_adjust_config_key_pair_types.h"
 
+#if defined(MBEDTLS_PSA_CRYPTO_C)
+/* If we are implementing PSA crypto ourselves, then we want to enable the
+ * required built-ins. Otherwise, PSA features will be provided by the server. */
 #include "mbedtls/config_adjust_legacy_from_psa.h"
+#endif
 
 #else /* MBEDTLS_PSA_CRYPTO_CONFIG */
 
-- 
2.39.2.windows.1

